aiger_bv/expr.py

```diff
@@ -33,6 +33,8 @@ def __getitem__(self, idx: Union[int, slice]):
         def _indexer(idx):
             return cmn.index_gate(self.size, idx, self.output, cmn._fresh())
 
+        if isinstance(idx, int) and idx < 0:
+            idx = idx % self.size
```

Comment:
Could you explain the motivation/use case here? Shouldn't we throw an error if `idx >= self.size`.

Reply:
Ah I see, this is for negative indexing.  Why check if `idx < 0`?

Perhaps instead:

```python
if isinstance(idx, int):
    assert -self.size <= idx < self.size
    idx %= self.size    # Allow for indexing from end of bit-vector.
```

Reply:
Good point. idx >= self.size should throw an error.

I'll implement it with a raise ValueError instead, if that's okay.

Reply:
Sounds good!